### ISSUE TITLE
Empty url in CSS

### ISSUE THREAD

**Issue by cristian-ungureanu** [Codeinwp/otter-blocks#690]
When a user selects a background for a column but doesn't add any image for it, the generated CSS looks like this:
`background: url() repeat scroll top left/auto;`

For some reason ( I can't explain exactly why ), in combination with WooCommerce, Neve's variation swatches on shop and custom layouts build with Otter, generates some kind of redirection when adding a product to the cart, and it adds the product to the cart twice. https://vertis.d.pr/njoc0A

How to replicate:
- Install Neve and Neve Pro
- Import Print Shop starter site
- After importing the Print Shop starter site, you'll notice that it adds a Custom layout for the header on the shop page build with Otter.
- Enable variation swatches on the shop page ( https://docs.themeisle.com/article/1359-variation-swatches-for-woocommerce-in-neve )
- Try to add a product to the cart, the behavior is either this: https://vertis.d.pr/njoc0A, or it adds just one product and after you leave the page it adds another.
- Remove the background selection https://vertis.d.pr/dlexh0 or https://vertis.d.pr/mNhndA and the issue is fixed

This issue is related to https://github.com/Codeinwp/neve-pro-addon/issues/1825

To summarize, we should check if a background (color or image ) is selected before adding the rule in CSS.

**Comment by HardeepAsrani** [comment]
Can you take a look? @Soare-Robert-Daniel

**Comment by cristian-ungureanu** [comment]
Can't we just check if there's a valid URL and if not to exclude the background CSS rule?

**Comment by cristian-ungureanu** [comment]
The template is fixed but this would happen for every template where the background is set like an image but no image is uploaded. We have to fix the problem from the root.

Can you better explain the issue and why do you think it's complicated to fix? I didn't exactly understand.

**Comment by Soare-Robert-Daniel** [comment]
You make a request to the page, the PHP triggers the CSS generation. This CSS comes as inline. If we a serious checking with a ping it will cause a delay. Also if we a ping, we might trigger some unexpected behavior if the link is bad.

A simple solution is to check the extension of the link, for valid image format. But I do not know if this method is fool-proof and with no edge cases.



https://github.com/Codeinwp/otter-blocks/blob/09a80537db8a5232f1bfcfa3e05d8c1793a1e613/inc/css/blocks/class-advanced-columns-css.php#L149-L176

**Comment by cristian-ungureanu** [comment]
Thank you @Soare-Robert-Daniel for the details! Let's see what @HardeepAsrani thinks.

**Comment by HardeepAsrani** [comment]
Yea, I've merged the PR. It seems enough to check for a valid image pattern. Thanks, Robert!
